### ISSUE TITLE
Curate challenges

### ISSUE THREAD

**Issue by byorgey** [swarm-game/swarm#1200]
Currently the Challenges menu seems a bit scattered.  For example, it starts with `Chess Knight` which is easy and silly, and then goes from there immediately to `A Frivolous Excursion` which is probably one of the hardest (well, longest, at least) challenges we have.  There's no particular rhyme or reason to the way things are organized (unless @kostmo has a secret organizational principle I haven't divined :smile: ).  This is no one's fault, just the natural result of adding challenges incrementally over time.

Better organization could include (1) putting the challenges in a better order, and/or (2) creating more categories/subfolders to put challenges in.

**Comment by kostmo** [comment]
I suppose this is a generalization of #1088.

**Comment by byorgey** [comment]
Here's my proposed categorization (just a draft, feel free to suggest changes):

- Programming
    - 2048
    - gopher
    - word-search
    - arbitrage
    - ice-cream
    - combo-lock
    - hackman
    - bucket-brigade.yaml
    - gallery.yaml
    - dimsum.yaml
- Logic
    - lights-out
    - wolf-goat-cabbage
    - hanoi
- RPG
    - bridge-building
    - blender.yaml
    - friend.yaml
- Mechanics
    - chess_horse
    - maypole
    - teleport
    - wave.yaml
    - pack-tetrominoes.yaml

**Comment by kostmo** [comment]
Perhaps "Programming" is overly general?  I'd suppose most of them involve programming.  "Algorithms" might be more specific.

Could have a category of "Themed puzzles", which captures most of what are currently in "Programming" (excluding `combo-lock`, `word-search`, `2048`).  Or we can have a subcategory "Environmental challenges", which may subsume the "Ranching" category.

**Comment by byorgey** [comment]
> Perhaps "Programming" is overly general?

Yes, you're probably right.  What I had in mind was scenarios where the main challenge/interest lies in writing a nontrivial program.  Not sure what to call that or whether it should even be its own category; I hope that category will continue to grow of course.

> Could have a category of "Themed puzzles" or "Story puzzles"

Sure, I like that idea.
